confmap/provider/secretsmanagerprovider/provider.go

```diff
@@ -24,6 +24,20 @@ type provider struct {
 //
 // This Provider supports "secretsmanager" scheme, and can be called with a selector:
 // `secretsmanager:NAME_OR_ARN`
+func NewFactory() confmap.ProviderFactory {
```

Comment:
The comment above this method needs to be updated to say `NewFactory` instead of `New`, and the return type should be updated as well.